app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/index.tsx

```diff
@@ -0,0 +1,218 @@
+  }
+`;
+
+const SelectColumnWrapper = styled(Box)`
+  height: 90%;
+`;
+
+const SelectColumnInnerWrapper = styled(Box)`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 10px;
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+const SelectColumnSearchBox = styled(Box)`
+  position: relative;
+  display: flex;
+`;
+
+const SearchIconButton = styled(IconButton)`
+  padding: 5px 0px 5px 5px;
+
+  &.MuiIconButton-root:hover {
+    background-color: transparent;
+  }
+  & .MuiTouchRipple-root {
+    display: none;
+  }
+`;
+
+export default function({
+  transformationDataType,
+  selectedColumnsCount,
+  columnsList,
+  setSelectedColumns,
+  dataQuality,
+  transformationName,
+  selectedColumns,
+}: ISelectColumnsListProps) {
+  const [columns, setColumns] = useState<IHeaderNamesList[]>(columnsList);
+  const [isSingleSelection, setIsSingleSelection] = useState<boolean>(true);
+  const ref = useRef(null);
+
+  const columnsAsPerType = getColumnsSupportedType(transformationDataType, columnsList);
+  const filteredColumnsOnType = getFilteredColumn(transformationDataType, columnsList);
+
+  useEffect(() => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN?.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName
+    );
+
+    if (multiSelect.length) {
+      setIsSingleSelection(false);
+    }
+    setColumns(filteredColumnsOnType);
+  }, []);
+
+  const onSingleSelection = (column: IHeaderNamesList) => {
+    setSelectedColumns([column]);
+  };
+
+  const onMultipleSelection = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    column: IHeaderNamesList
+  ) => {
+    if (event.target.checked) {
+      setSelectedColumns((prev) => [...prev, column]);
+    } else {
+      const indexOfUnchecked = selectedColumns.findIndex(
+        (columnDetail) => columnDetail.label === column.label
+      );
+      if (indexOfUnchecked > -1) {
+        setSelectedColumns(() => selectedColumns.filter((_, index) => index !== indexOfUnchecked));
+      }
+    }
+  };
+
+  const handleDisableCheckbox = () => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName && functionDetail.isMoreThanTwo
+    );
+    if (selectedColumns?.length === 0 || selectedColumns?.length < 2) {
+      return false;
+    } else if (multiSelect.length) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value) {
+      const columnValue: IHeaderNamesList[] = filteredColumnsOnType.length
+        ? filteredColumnsOnType.filter((columnDetail: IHeaderNamesList) =>
```

Comment:
no multiline ternary operators
plz update this code asap
client has been pointing this since beginning

plz check across this PR and other PRs and fix this asap

Reply:
Done